coalib/misc/Caching.py

```diff
@@ -173,3 +178,137 @@ def get_uncached_files(self, files):
+    >>> proxycache.set_proxymap(proxymap)
+
+    A file dict can now be build using the underlying proxy
+    map using:
+
+    >>> proxy = FileProxy.from_file(file.name, None)
+    >>> proxymap.add(proxy)
+    True
+
+    >>> file_dict = proxycache.get_file_dict([file.name])
+    >>> file_dict[file.name]
+    ('coala',)
+    """
+
+    def __init__(self, *args, **kargs):
+        """
+        This directly initializes the associated FileCache.
+        """
+        super().__init__(*args, **kargs)
+        self.__proxymap = None
+
+    @enforce_signature
+    def set_proxymap(self, fileproxy_map: FileProxyMap):
+        """
+        Used to assign a ProxyMap, this is separate from
+        the instance initialization method to keep the
+        concerns separate.
+
+        :param fileproxy_map:   FileProxyMap instance to build
+                                the file dict from.
+        """
+        self.__proxymap = fileproxy_map
+
+    def get_file_dict(self, filename_list, allow_raw_files=False):
+        """
+        Builds a file dictionary from filename to lines of the file
+        from an associated FileProxyMap.
+
+        :param filename_list:   List of files to get the contents of.
+        :param allow_raw_files: Allow the usage of raw files (non text files),
+                                disabled by default
+        :return:                Reads the content of each file into dictionary
+                                with filenames as keys.
+        """
+        if self.__proxymap is None:
+            raise ValueError('set_proxymap() should be called to set proxymap'
+                             'of ProxyMapFileCache instance')
+
+        file_dict = {}
+        for filename in filename_list:
+            try:
+                # Keep the binary mode turned off to maintain compatibility
+                # with FileDictFileCache.get_file_dict().
+                # filename is assumed to be normcased.
+                proxy = self.__proxymap.resolve(filename,
+                                                hard_sync=True,
+                                                binary=False)
+
+                file_lines = proxy.lines()
+                file_dict[filename] = tuple(file_lines)
+            except UnicodeDecodeError:
+                if allow_raw_files:
+                    file_dict[filename] = None
+                    continue
+
+                logging.warning("Failed to read file '{}'. It seems to contain "
+                                'non-unicode characters. Leaving it out.'
+                                .format(filename))
+
+            except (OSError, FileNameError) as exception:
+                log_exception("Failed to read file '{}' because of an unknown "
+                              'error. Leaving it out.'.format(filename),
+                              exception,
+                              log_level=LOG_LEVEL.WARNING)
+
+        logging.debug('Files that will be checked:\n' +
```

Comment:
this could be moved to the common path in `instantiate_processes` after the dict has been obtained.

Reply:
I think we need to maintain compatibility with `FileDictFileCache`, since it uses the default implementation of `get_file_dict()` it logs locally as soon as files are collected. Moving this to `instantiate_processes()` will cause logging twice for FileDictFileCache. I think we can have a separate issue for this once this gets merged.

Reply:
I mean that `get_file_dict` should also not log this message.

`instantiate_processes` can do that for all implementations.